OpenSim/Common/Exception.h

```diff
@@ -67,14 +110,31 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
```

Comment:
It is necessary to allow changing the delimiter. Being able to change the delimiter from "\n" to ":" if desired is a reasonable justification to have it there. If we implement the formatting of message differently in future, this constant can be removed at that point. I don't see this constant as causing any problem now or later.
The constant is not part of public interface. It is accessible only to derived classes in case they want to use the delimiter in formatting their own messages.

I am not sure I understand the problem about this constant and if it is even worth this much of discussion.

I did not want to change the access specifiers for the existing members in Exception class. This PR is not about editing Exception. We should do that in another PR if desired.